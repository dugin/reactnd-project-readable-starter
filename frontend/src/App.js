import React, {Component} from 'react';
import './App.css';
import Header from "./components/header/Header";
import Footer from "./components/footer/Footer";
import ReadablePage from "./components/readable-page/Readable.page";
import {Switch, Route, Redirect, withRouter} from 'react-router-dom';
import CreateEditPage from "./components/create-edit-page/CreateEdit.page";
import DetailPage from "./components/detail-page/Detail.page";
import {connect} from "react-redux";
import {fetchPosts} from "./actions/posts";
import {fetchCategories} from "./actions/categories";

class App extends Component {

    componentDidMount() {
        this.props.dispatch(fetchPosts());
        this.props.dispatch(fetchCategories());
    }

    render() {
        return (
            <div className="App">
                <Header/>
                <main className="App__main container">
                    <Switch>
                        <Route exact path="/" component={ReadablePage}/>
                        <Route exact path="/create" component={CreateEditPage}/>
                        <Route exact path="/:category" component={ReadablePage}/>
                        <Route exact path="/edit/:post_id" component={CreateEditPage}/>
                        <Route exact path="/:category/:post_id" component={DetailPage}/>
                        <Redirect to="/" path="**" component={ReadablePage}/>
                    </Switch>
                </main>
                <Footer/>
            </div>
        );
    }
}

export default withRouter(connect(null)(App));
