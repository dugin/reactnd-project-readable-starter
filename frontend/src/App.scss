.App {
  height: 98%;
  position: relative;


  &__main{
    min-height: calc(100% - 100px);
    display: flex;
    flex-direction: column;
  }
}
