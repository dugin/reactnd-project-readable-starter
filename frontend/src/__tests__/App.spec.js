import React from 'react';
import App from "../App";
import {shallow} from "enzyme";

describe('App Component', () => {

    it('renders without crashing', () => {
       const app = shallow(<App/>);

       expect(app).toMatchSnapshot()

    });


});
