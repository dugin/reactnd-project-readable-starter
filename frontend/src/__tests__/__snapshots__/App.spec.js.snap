// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders properly 1`] = `
<div
  className="App"
>
  <withRouter(Header) />
  <main
    className="App__main container"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/create"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/edit"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/post/:id"
      />
      <Redirect
        component={[Function]}
        path="**"
        push={false}
        to="/"
      />
    </Switch>
  </main>
  <Footer />
</div>
`;
